### DISCUSSION TITLE
_autoAction

### DISCUSSION THREAD

**Discussion by blitzbohne** [mobxjs/mobx#2635] [Q&A]
Hello everyone,

We're having our own method that is similar to makeAutoObservable. We'd like to use the new autoAction but it seems the mobx exports it as "_autoAction" instead of "autoAction"? Is there any reason for this? Is it safe to use?

**Comment by urugator** [comment]
Probably is, if you want to be sure you can use less efficient variant `makeAutoObservable({ fn }).fn`.
There is also a possibility, that `observable(fn)` will return auto-action, but I can't promise that atm.

**Comment by mweststrate** [comment]
It should be fine to use it in libraries, the `_`indicates primarily indicates that it shouldn't be used directly by consumer code. So that, if behavior or name changes in the future, you'll have only a single place to update
